languages/scala/reference/concept-exercises/scrabble-score.md

```diff
@@ -0,0 +1,25 @@
+# Concepts of scrabble-score
+
+[Example implementation](https://github.com/exercism/scala/blob/master/exercises/scrabble-score/example.scala)
+
+## General
+
+- object or case class: used as class wrapper for exercises
+- methods: used as main entry point for the exercise and helper methods
+- functions: each method is a function
+- function arguments: input strands passed as arguments
+- inline functions: a lambda is defined in the `flatMap` call
+- return values: return a value from a function
+- visibility: making values `private`
+- scoping: single-line function scoping
+- immutability: define immutable values using `val`
+- pattern matching: a pattern match is used in the lambda
+- math operators: `+` to calculate total score
+- collections - folds: a fold is used to calculate the total score
+- for-comprehension: a for-comprehensen is used to create a letter scores map
+- integers: an `Int` is used to represent a letter score and the total score
+- characters: a `Char` is used to represent a letter
+- strings: a `String` is used to represent the input word
+- dictionaries: a `Map[Char, Int]` is used for the return value
```

Comment:
Maybe worth talking about the `getOrElse` since it will come back with `Option`

Reply:
Is `getOrElse` a separate concept though?

Reply:
Or do you mean: add it to the description?

Reply:
something like:

```
collection - accessors: getOrElse: retrieve a value given a key or use default value
```

I think it is important because the `getOrElse` function comes back many times in Scala with Either and Option , etc...

Reply:
Hmmm, let me think on this a little more!

Reply:
@ricemery What do you think?

Reply:
I think I agree with @leobenkel . `getOrElse` is one of those things that might not be seen/understood by a student.  But, once it is seen the concept is very useful in simplifying code. So, calling it out somehow I think would be useful.

Reply:
Alright. Let's include it then.

Reply:
Updated.